tests/backwards_compatibility_tests/compatibility_test_runner.py

```diff
@@ -615,6 +525,188 @@ def copy_state_from_container(
+
+def backwards_compatibility_test(from_version: str, to_version: str, to_version_image_idenfitifer: str):
+    """
+    Perform a backwards compatibility test between two versions of Marqo.
+
+    This function starts a container with the from_version, runs tests in prepare mode, stops the container,
+    starts a container with the to_version by transferring state from from_version container, and runs tests in test mode.
+
+    Args:
+        from_version (str): The source version of the Marqo container.
+        to_version (str): The target version of the Marqo container.
+        to_version_image_idenfitifer (str): The unique identifier for a to_version image. It can be either be the fully qualified image name with the tag
+                                (ex: 424082663841.dkr.ecr.us-east-1.amazonaws.com/marqo-compatibility-tests:abcdefgh1234)
+                                or the fully qualified image name with the digest (ex: 424082663841.dkr.ecr.us-east-1.amazonaws.com/marqo-compatibility-tests@sha256:1234567890abcdef).
+                                This is constructed in build_push_image.yml workflow and will be the qualified image name with digest for an automatically triggered workflow.
+
+    Raises:
+        ValueError: If the major versions of from_version and to_version are incompatible.
+        Exception: If there is an error during the test process.
+    """
+    try:
+        # Step 1: Start from_version container and run tests in prepare mode
+        logger.debug(f"Starting backwards compatibility tests with from_version: {from_version}, to_version: {to_version}, to_version_image_idenfitifer: {to_version_image_idenfitifer}")
+
+        # Generate a volume name to be used with the "from_version" Marqo container for state transfer.
+        from_version_volume = get_volume_name_from_marqo_version(from_version)
+
+        #Start from_version container
+        start_marqo_container(from_version, from_version_volume)
+        logger.debug(f"Started marqo container {from_version}")
+
+        try:
+            run_tests_across_versions("prepare", from_version)
+        except Exception as e:
+            logger.error(f"Error running tests across versions in 'prepare' mode: {e}")
+            raise e
+        # Step 2: Stop from_version container (but don't remove it)
+        stop_marqo_container(from_version)
+
+        # Step 3: Start to_version container by transferring state
+        logger.debug(f"Starting marqo to_version: {to_version} container by transferring state from version {from_version} to {to_version}")
+        start_marqo_container_by_transferring_state(to_version, from_version, from_version_volume,
+                                                    to_version_image_idenfitifer, "ECR")
+        logger.debug(f"Started marqo to_version: {to_version} container by transferring state")
+        # Step 4: Run tests
+        try:
+            run_tests_across_versions("test", from_version)
```

Comment:
Can we have an enum for mode? Passing strings around is error prone and confusing